tests/cpp-tests/Classes/UITest/CocoStudioGUITest/UITextFieldTest/UITextFieldTest.cpp

```diff
@@ -416,6 +417,82 @@ void UITextFieldTest_TrueTypeFont::textFieldEvent(Ref *pSender, TextField::Event
     }
 }
 
+// UITextFieldTest_BMFont
+UITextFieldTest_BMFont::UITextFieldTest_BMFont()
+: _displayValueLabel(nullptr)
+{
+    
+}
+
+UITextFieldTest_BMFont::~UITextFieldTest_BMFont()
+{
+}
+
+bool UITextFieldTest_BMFont::init()
+{
+    if (UIScene::init())
+    {
+        Size widgetSize = _widget->getContentSize();
+        
+        // Add a label in which the textfield events will be displayed
+        _displayValueLabel = Text::create("BMFont Test - No Event","fonts/Marker Felt.ttf",32);
+        _displayValueLabel->setAnchorPoint(Vec2(0.5f, -1.0f));
+        _displayValueLabel->setPosition(Vec2(widgetSize.width / 2.0f, widgetSize.height / 2.0f + _displayValueLabel->getContentSize().height * 1.5f));
+        _uiLayer->addChild(_displayValueLabel);
+        
+        // Add the alert
+        Text* alert = Text::create("TextField","fonts/Marker Felt.ttf",30);
+        alert->setPosition(Vec2(widgetSize.width / 2.0f, widgetSize.height / 2.0f - alert->getContentSize().height * 3.075f));
+        _uiLayer->addChild(alert);
+        
+        // Create the textfield
+        TextField* textField = TextField::create("input words here","fonts/markerFelt.fnt",25);
```

Comment:
suggestion 1st:
```
        // Create the textfield
        TextField* textField = TextField::create("input words here","fonts/bitmapFontTest3.fnt",25);
        textField->setCursorEnabled(true);
```
1. `fonts/bitmapFontTest3.fnt` is more colorful, it's better to show this feature, right?
1.  add `setCursorEnabled(true);` to let developers know BMFont supports cursor too. I have tested it, works well.

Reply:
Will have a look at this. I think it has something to do with the position in the update method.